### ISSUE TITLE
CA1062 should not warn here #2

### ISSUE THREAD

**Issue by matjazmav** [dotnet/roslyn-analyzers#2636]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Package Version

2.9.2 & 2.9.3 (tried both)

#### Diagnostic ID

CA1062

#### Repro steps

We have static helper fuction that checks for null reference, this worked with deprecated CA:
```
        [DebuggerStepThrough]
        [ContractAnnotation("value:null => halt")]
        [NotNull]
        public static T NotNull<T>([ValidatedNotNull][NoEnumeration] T value)
        {
            if (!ReferenceEquals(value, null))
                return value;

            throw new ArgumentException("Input cannot be null.");
        }
```

We use this helper like that:
```
        public static int CalculateAnnualDecisionUsedDays(List<CollaborationLeave> annualDecisions, ...)
        {
            Check.NotNull(annualDecisions);

            foreach (var ad in annualDecisions/*here is error CA1062*/)
            {
                var dm = ad.DateModified;
            }

           // ...
```

#### Expected behavior
Object is checked for null, no error shuld raised.

#### Actual behavior
CA1062 error is raised.

**Comment by matjazmav** [comment]
Ok it works on 2.9.4-beta1.final ...
